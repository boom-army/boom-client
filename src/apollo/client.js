import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
