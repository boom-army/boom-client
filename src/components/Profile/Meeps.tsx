import React from "react";
import { Box, Typography } from "@mui/material";
import { ShowTweet } from "../Tweet";
import { Tweet, ProfileQuery } from "../../generated/graphql";

export const Meeps: React.FC<{
  data: ProfileQuery | undefined;
}> = ({ data }) => {
  return (
    <>
      {data && data.profile && data.profile.tweets && data.profile.tweets.length
        ? data.profile.tweets.map((tweet) => (
            <ShowTweet
              key={tweet.id}
              tweet={tweet as Tweet}
              offset={10}
              parentTweetId=""
            />
          ))
        : (
          <Box pt={10} sx={{ display: "flex", justifyContent: "center" }}>
            <Typography>You haven't Meeped yet</Typography>
          </Box>
        )}
    </>
  );
};
