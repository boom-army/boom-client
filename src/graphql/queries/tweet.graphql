query tweet($id: ID!) {
  tweet(id: $id) {
    ...TweetData
    masterTweets {
      ...TweetData
    }
  }
}

mutation newTweet(
  $text: String!
  $files: [String!]!
  $tags: [String!]!
  $mentions: [String!]!
  $gif: GifInput
  $nft: NFTInput
  $parentTweet: String
  $masterTweet: String
  $channel: String
) {
  newTweet(
    text: $text
    files: $files
    tags: $tags
    mentions: $mentions
    gif: $gif
    nft: $nft
    parentTweet: $parentTweet
    masterTweet: $masterTweet
    channel: $channel
  ) {
    id
    text
    tags
    mentions
    commentsCount
    createdAt
  }
}

mutation toggleRetweet($id: ID!) {
  toggleRetweet(id: $id)
}

mutation deleteTweet($id: ID!) {
  deleteTweet(id: $id) {
    id
  }
}
