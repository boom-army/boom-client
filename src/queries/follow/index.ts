import gql from "graphql-tag";

export const FOLLOW = gql`
  mutation follow($id: ID!) {
    follow(id: $id)
  }
`;

export const UNFOLLOW = gql`
  mutation unfollow($id: ID!) {
    unfollow(id: $id)
  }
`;

export const USERS = gql`
  query users($offset: Int, $limit: Int, $filter: String) {
    users(offset: $offset, limit: $limit, filter: $filter) {
      id
      handle
      isFollowing
      isSelf
      consumerName
      avatar
      data {
        avatarMint
        avatarUpdateAuthority
      }
      coverPhoto
      bio
    }
  }
`;

export const USER_FOLLOW = gql`
  query userFollow($offset: Int, $limit: Int) {
    userFollow(offset: $offset, limit: $limit) {
      id
      handle
      avatar
      data {
        avatarMint
        avatarUpdateAuthority
      }
      isFollowing
      isSelf
      consumerName
      bio
    }
  }
`;
