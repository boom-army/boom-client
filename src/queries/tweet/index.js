import gql from "graphql-tag";

export const NEW_TWEET = gql`
  mutation newTweet(
    $text: String!
    $files: [String!]!
    $tags: [String!]!
    $mentions: [String!]!
    $gif: GifInput
    $nft: NFTInput
    $parentTweet: String
  ) {
    newTweet(
      text: $text
      files: $files
      tags: $tags
      mentions: $mentions
      gif: $gif
      nft: $nft
      parentTweet: $parentTweet
    ) {
      id
      text
      tags
      mentions
      commentsCount
      createdAt
    }
  }
`;

export const DELETE_TWEET = gql`
  mutation deleteTweet($id: ID!) {
    deleteTweet(id: $id) {
      id
    }
  }
`;

export const TOGGLE_RETWEET = gql`
  mutation toggleRetweet($id: ID!) {
    toggleRetweet(id: $id)
  }
`;

export const TOGGLE_REACTION = gql`
  mutation toggleReaction($id: ID!, $emojiId: String!, $skin: Int) {
    toggleReaction(id: $id, emojiId: $emojiId, skin: $skin)
  }
`;

const TWEET_FRAGMENT = gql`
  fragment subTweet on Tweet {
    id
    text
    tags
    mentions
    retweetsCount
    isRetweet
    createdAt
    parentTweet {
      id
    }
    user {
      id
      publicAddress
      consumerName
      handle
      avatar
    }
    files {
      id
      url
    }
    gif {
      id
      title
      fixedHeightUrl
      originalUrl
    }
    nft {
      id
      publicKey
      name
      symbol
      description
      sellerFeeBasisPoints
      externalUrl
      image
      attributes {
        traitType
        value
      }
      collection {
        name
        family
      }
      properties {
        files {
          uri
          type
        }
        category
        creators {
          address
          share
        }
      }
    }
    reactions {
      id
      emojiId
      skin
      isMine
      count
    }
  }
`;

export const TWEET = gql`
  query tweet($id: ID!) {
    tweet(id: $id) {
      ...subTweet
      childTweets {
        ...subTweet
        childTweets {
          ...subTweet
        }
      }
    }
  }
  ${TWEET_FRAGMENT}
`;
