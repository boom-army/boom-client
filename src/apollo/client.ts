import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
