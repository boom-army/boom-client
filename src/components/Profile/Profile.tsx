import React, { useEffect, useState } from "react";
import ProfileInfo from "./ProfileInfo";
import { Box, Tab } from "@mui/material";
import { Loader } from "../Loader";
import { Meeps } from "./Meeps";
import { TabContext, TabPanel, TabList } from "@mui/lab";
import { useParams } from "react-router-dom";
import { useProfileLazyQuery } from "../../generated/graphql";
import { NFTGallery } from "./NFTGallery";
import { CustomResponse } from "../CustomResponse";

export const Profile: React.FC = () => {
  const [tabValue, setTabValue] = useState("1");

  let { handle } = useParams<string>();

  const [setHandle, { loading, data }] = useProfileLazyQuery();

  useEffect(() => {
    if (handle) setHandle({ variables: { handle } });
  }, [handle, setHandle]);

  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setTabValue(newValue);
  };

  if (loading) {
    return <Loader />;
  }
  if (!loading && !data) {
    return (
      <CustomResponse text="Oops, you are trying to visit a profile which doesn't exist. Make sure the profile handle exists" />
    );
  }

  return (
    <>
      <ProfileInfo profile={data && data.profile} />
      <Box sx={{ typography: "body1" }}>
        <TabContext value={tabValue}>
          <Box sx={{ borderBottom: 2, borderColor: "#374148" }}>
            <TabList
              onChange={handleChange}
              aria-label="Profile tablist select"
            >
              <Tab sx={{ minWidth: 150 }} label="NFT Gallery" value="1" />
              <Tab label="Meeps" value="2" />
            </TabList>
          </Box>
          <TabPanel value="1">
            {data && <NFTGallery publicAddress={data?.profile.publicAddress} />}
          </TabPanel>
          <TabPanel value="2">
            <Meeps data={data} />
          </TabPanel>
        </TabContext>
      </Box>
    </>
  );
};
