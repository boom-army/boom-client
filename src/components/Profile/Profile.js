import React from "react";
import { useQuery } from '@apollo/client';
import { useParams } from "react-router-dom";
import styled from "styled-components";
import ProfileInfo from "./ProfileInfo";
import { ShowTweet } from "../Tweet";
import { Loader } from "../Loader";
import { PROFILE } from "../../queries/profile";

const Wrapper = styled.div`
	padding-bottom: 5rem;

  .profile-top {
    display: flex;
    flex-direction: column;
    margin-left: 1rem;

    span.tweetsCount {
      margin-top: 0.1rem;
      color: ${(props) => props.theme.secondaryColor};
      font-size: 0.9rem;
    }
  }
`;

export const Profile = () => {
  const { handle } = useParams();

  const { loading, error, data } = useQuery(PROFILE, {
    variables: { handle },
  });

  if (loading) return <Loader />;

  return (
    <Wrapper>
      {/* <div className="profile-top">
        <span>{data && data.profile && data.profile.consumerName}</span>
        <span className="tweetsCount">
          {data && data.profile && data.profile.tweetsCount
            ? `${data.profile.tweetsCount} Tweets`
            : "No Tweets"}
        </span>
      </div> */}
      <ProfileInfo profile={data && data.profile} />
      {data && data.profile && data.profile.tweets && data.profile.tweets.length
        ? data.profile.tweets.map((tweet) => (
          <ShowTweet key={tweet.id} tweet={tweet} />
        ))
        : null}
    </Wrapper>
  );
};
