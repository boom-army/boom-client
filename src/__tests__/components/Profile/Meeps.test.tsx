import { Meeps } from "../../../components/Profile/Meeps";
// import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { FOLLOW } from "../../../queries/follow/index";
import { TWEET } from "../../../queries/tweet/index";
// import { Tweet } from "../../../generated/graphql";
import { act, fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SnackbarProvider } from "notistack";

const profile: any = {
  data: {
    profile: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      avatar: "",
      coverPhoto: "/default-cover.png",
      dob: "",
      location: "",
      website: "",
      isSelf: true,
      isFollowing: false,
      followersCount: 0,
      followingCount: 0,
      tweetsCount: 21,
      newMentionsCount: 0,
      bio: "full stack developer",
      tweets: [
        {
          id: "cl0c77yow71753s0n5h032nerw",
          text: "good scope.",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646385168992",
          __typename: "Tweet",
        },
        {
          id: "cl084xdy815761s0n5omoxoem0",
          text: "testing after changes in input converted to ts",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [
            {
              id: "cl0c774p371312s0n533u0b29k",
              emojiId: "+1",
              skin: 1,
              isMine: true,
              count: 1,
              __typename: "Reaction",
            },
          ],
          commentsCount: 1,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646139451616",
          __typename: "Tweet",
        },
        {
          id: "cl081ymfz8867s0n59y782hdj",
          text: "🥰",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646134470431",
          __typename: "Tweet",
        },
        {
          id: "cl06ku5qr9948ban5356ugu9b",
          text: "😍",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646045242515",
          __typename: "Tweet",
        },
        {
          id: "ckzzg3lm6169405vun5im00fsmq",
          text: "😘6210",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1645614021678",
          __typename: "Tweet",
        },
      ],
      createdAt: "1644918563773",
      __typename: "User",
    },
  },
};

// export const __mocks__: any = [
//   {
//     request: {
//       query: FOLLOW,
//       variables: {
//         id: "ckzny1iv10031lmn568gj3b3n",
//       },
//     },
//     result: {
//       data: {
//         mentions: [],
//       },
//       errors: "An error occurred",
//     },
//   },
// ];

export const __mocks__: any = [
  {
    request: {
      query: TWEET,
      variables: {
        // id: "ckzny1iv10031lmn568gj3b3n",
        id: "cl0c77yow71753s0n5h032nerw"
      },
    },
    result: {
      data: {
        mentions: [],
      },
      errors: "An error occurred",
    },
  },
];

// describe("Meeps : ", () => {
//   test("renders correctly user Meeps", () => {
//     const tree = renderer.create(<Meeps data={profile.data} />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   test("User haven't Meeped yet", () => {
//     const tree = renderer.create(<Meeps data={undefined} />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });

test("it should ...", async () => {
  let rendered;
  await act(async () => {
    rendered = render(
      <MockedProvider mocks={__mocks__} addTypename={false}>
      <SnackbarProvider>
       <Meeps data={profile.data} />
       </SnackbarProvider>
      </MockedProvider>,{wrapper: MemoryRouter}
    );
  });
  expect(rendered).toMatchSnapshot();
});

