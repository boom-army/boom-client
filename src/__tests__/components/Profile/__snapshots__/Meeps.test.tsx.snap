// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meeps :  User haven't Meeped yet 1`] = `
<div
  className="MuiBox-root css-uz95m6"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    You haven't Meeped yet
  </p>
</div>
`;
